src/com/jaquadro/minecraft/storagedrawers/block/BlockDrawers.java

```diff
@@ -415,94 +413,66 @@ protected boolean hitLeft (int side, float hitX, float hitZ) {
-        {
-            @Override
-            public void run () {
-                BlockDrawers.this.onBlockDestroyedAsync(world, pos);
-            }
-        });
-    }
-
-    private void onBlockDestroyedAsync (World world, BlockPos pos) {
-        //this.onBlockHarvested(world, pos, state, player);
-        world.setBlockState(pos, net.minecraft.init.Blocks.AIR.getDefaultState(), world.isRemote ? 11 : 3);
-    }
-
-    @Override
-    public void onBlockClicked (World world, BlockPos pos, EntityPlayer player) {
-        if (world.isRemote) {
-            RayTraceResult ray = Minecraft.getMinecraft().objectMouseOver;
-            BlockPos posb = ray.getBlockPos();
-            float hitX = (float)(ray.hitVec.xCoord - posb.getX());
-            float hitY = (float)(ray.hitVec.yCoord - posb.getY());
-            float hitZ = (float)(ray.hitVec.zCoord - posb.getZ());
-
-            StorageDrawers.network.sendToServer(new BlockClickMessage(pos.getX(), pos.getY(), pos.getZ(), ray.sideHit.ordinal(), hitX, hitY, hitZ, StorageDrawers.config.cache.invertShift));
-
-            if (StorageDrawers.config.cache.debugTrace)
-                FMLLog.log(StorageDrawers.MOD_ID, Level.INFO, "BlockDrawers.onBlockClicked with " + ray.toString());
         }
-    }
 
-    @Override
-    public void onBlockClicked (final World world, final BlockPos pos, final EntityPlayer player, final EnumFacing side, final float hitX, final float hitY, final float hitZ, final boolean invertShift) {
-        if (world.isRemote)
-            return;
-
-        ((WorldServer)world).addScheduledTask(new Runnable()
-        {
-            @Override
-            public void run () {
-                BlockDrawers.this.onBlockClickedAsync(world, pos, player, side, hitX, hitY, hitZ, invertShift);
-            }
-        });
-    }
-
-    private void onBlockClickedAsync (World world, BlockPos pos, EntityPlayer player, EnumFacing side, float hitX, float hitY, float hitZ, boolean invertShift) {
         if (StorageDrawers.config.cache.debugTrace)
-            FMLLog.log(StorageDrawers.MOD_ID, Level.INFO, "IExtendedBlockClickHandler.onBlockClicked");
+            FMLLog.log(StorageDrawers.MOD_ID, Level.INFO, "onBlockClicked");
+
+        RayTraceResult rayResult = net.minecraftforge.common.ForgeHooks.rayTraceEyes(playerIn, ((EntityPlayerMP) playerIn).interactionManager.getBlockReachDistance() + 1);
+        EnumFacing side = rayResult.sideHit;
+        // adjust hitVec for drawers
+        float hitX = (float)(rayResult.hitVec.xCoord - pos.getX());
+        float hitY = (float)(rayResult.hitVec.yCoord - pos.getY());
+        float hitZ = (float)(rayResult.hitVec.zCoord - pos.getZ());
 
-        if (!player.capabilities.isCreativeMode) {
-            PlayerInteractEvent.LeftClickBlock event = new PlayerInteractEvent.LeftClickBlock(player, pos, side, new Vec3d(hitX, hitY, hitZ));
+        if (!playerIn.capabilities.isCreativeMode) {
```

Comment:
@Zidane I'll remove it but there is a case where onBlockClicked is called if a player attempts to break the front side of a storage block. However, you are correct that the event should fire normally for the actual break so it shouldn't be needed. I'll test it tonight.